dom/src/main/java/org/apache/james/mime4j/internal/ParameterHelper.java

```diff
@@ -0,0 +1,66 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mime4j.internal;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```

Comment:
This is a blocker for me. Java regex performance is terrible. I prefer 100x avoiding it.

Reply:
can't we just javacc here ?

Reply:
Looks legitimate to me.

Reply:
> This is a blocker for me. Java regex performance is terrible. I prefer 100x avoiding it. And write plain java code (we can back it up with something like https://github.com/CodeIntelligenceTesting/jazzer )

I just have to ask; This used to be an issue "back in the days", but do you have any documentation supporting the regexps used here are actually that slow and pose any risks? I think on any modern hardware this is a none-issue and a waste of effort trying to do with non-regexps. Using javacc for this is IMHO overkill, and also might actually be slower, for such simple cases as we don't need a full parser for this.

Reply:
https://github.com/apache/james-mime4j/pull/50 is such an example.

https://github.com/apache/james-mime4j/commit/66a87ba9b1828b57618066ec682c8a11f34ad413 another.

To be fair #50 was worth it for users (like some part of the ames project) aggressively decoding every fields.

Reply:
Ok, thanks. I still think javacc is overkill for this task and suggest to just parse this manually.

Reply:
I'm ok with manual parsing. I even propose mysel to run Jazzer against the resulting code if it helps.

(I need to do a Jazzer demo at the office, this could be a nice convergence of topics ;-) )